src/screens/Login/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { Auth } from "aws-amplify";
+import React, { Component } from "react";
+import { ControlLabel, FormControl, FormGroup } from "react-bootstrap";
+import LoaderButton from "../../components/LoaderButton";
+import "./index.css";
+
+class Login extends Component {
+  state = {
+    isLoading: false,
+    email: "",
+    password: "",
+  };
+
+  componentDidMount() {
+    this.props.isAuthenticated && this.props.history.push('/');
+  }
+
+  validateForm = () => (this.state.email.length && this.state.password.length)
+
+  handleChange = event => {
+    this.setState({
+      [event.target.id]: event.target.value
+    });
+  };
+
+  handleSubmit = async event => {
+    event.preventDefault();
+
+    this.setState({ isLoading: true });
+
+    const { email: username, password } = this.state;
+
+    try {
+      await Auth.signIn({ username, password });
+
+      this.handleLoginSuccess();
+    } catch (e) {
+      alert(e.message);
+    }
+
+    this.setState({ isLoading: false });
```

Comment:
I totally agree with you and I also know the idea about `callback` of `this.setState`
But, in my opinion, It will be best practices any time you use the `state` between the *chain*
like this

```
this.setState({ count: 1 }, () => {
  ...
  const nextCount = this.state.count + 1;
  ...
  this.setState({ count: nextCount }); // absolutely 2
});
```
Otherwise, just write it normally will be easier to read

Reply:
If your middle block introduce more action to redux later on, it may cause UI change in middle as well. Put it inside a call back will make sure your UI flow is always consistent regardless any code change later on.

Reply:
Okay, I get it.
I have lack low-tier mobile experience